pkg/payload/payload.go

```diff
@@ -259,20 +259,18 @@ func GetImplicitlyEnabledCapabilities(updatePayloadManifests []manifest.Manifest
 			if err := currentManifest.IncludeAllowUnknownCapabilities(nil, nil, nil, &clusterCaps, nil, true); err != nil {
 				continue
 			}
-			caps := capability.GetImplicitlyEnabledCapabilities(currentManifest.GetManifestCapabilities(),
-				updateManifest.GetManifestCapabilities(), capabilities)
-
-			capStrings := make([]string, len(caps))
-			for i, c := range caps {
-				capStrings[i] = string(c)
-				if !capability.Contains(implicitlyEnabledCaps, c) {
-					implicitlyEnabledCaps = append(implicitlyEnabledCaps, c)
-				}
+			caps := capability.GetImplicitlyEnabledCapabilities(sets.New[configv1.ClusterVersionCapability](currentManifest.GetManifestCapabilities()...),
+				sets.New[configv1.ClusterVersionCapability](updateManifest.GetManifestCapabilities()...), capabilities)
+			implicitlyEnabledCaps = implicitlyEnabledCaps.Union(caps)
+			if caps.Len() > 0 {
+				klog.V(2).Infof("%s has changed and is now part of one or more disabled capabilities. The following capabilities will be implicitly enabled: %s",
+					getManifestResourceId(updateManifest), capability.SortedList(caps))
 			}
-			klog.V(2).Infof("%s has changed and is now part of one or more disabled capabilities. The following capabilities will be implicitly enabled: %s",
-				getManifestResourceId(updateManifest), strings.Join(capStrings, ", "))
 		}
 	}
+	if implicitlyEnabledCaps.Len() == 0 {
+		return nil
+	}
```

Comment:
This seems not needed.
I do not think it is worth another round of CI.
I will handle it later.